### PR TITLE
clean-vpc-peering-when-deleting-cluster

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/15647

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/resource/cleanupvpcpeerings/delete.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+				vpcPeeringConnectionIDs = append(vpcPeeringConnectionIDs, s.VpcPeeringConnectionId)
+			}
+		}
+		// fetch all vpc peering connection via accepter vpc filter
+		{
+			i := &ec2.DescribeVpcPeeringConnectionsInput{
+				Filters: []*ec2.Filter{
+					{
+						Name: aws.String("accepter-vpc-info.vpc-id"),
```
Comment (njuettner) [inline service/controller/resource/cleanupvpcpeerings/delete.go:67]: does it make sense do have it in a small helper function? it seems it's duplicate code except the name is different.

**Review by njuettner — COMMENTED** [review]
just a nit, not sure if that would make sense what I wrote. let me know

**Review by xh3b4sd — APPROVED** [review]
I am happy when Nick is happy.

File: service/controller/resource/cleanupvpcpeerings/delete.go (hunk 1 above)
Comment (calvix) [reply service/controller/resource/cleanupvpcpeerings/delete.go]: I have adjusted the code, can you check now?

**Review by njuettner — APPROVED** [review]
🚀

**Pull request merged**